### DISCUSSION TITLE
Cross validation (CV) for a model with year of AR(1)

### DISCUSSION THREAD

**Discussion by shyunuw** [sdmTMB/sdmTMB#547] [Q&A]
Hi.
The cross validation (cv) tool, "sdmTMB_cv" is useful, but it seems to be limited in a temporal autocorrelation model (AR) model.  For example, a candidate model for my work is as follows:

fit <- sdmTMB(y  ~ 1 + (1|random_effect),
                                  data = mydata,
                                  mesh = mymesh,
                                  family = tweedie(link="log"),
                                  offset = log(mydata$fishing_efforts),
                                  spatial="on",
                                  time="year", time_varying=~1, time_varying_type="ar1",
                                  spatiotemporal="iid"
                                 ); 

The above model has AR(1), and thus it is not appropriate to randomly select a training set and a validation set for CV.   I wonder whether or not an efficient way is implemented in the sdmTMB package.  If so, would you please guide me with such an efficient code?  Otherwise, I *manually* plan to select training and validation sets for CV, considering the temporal order in data.  I would greatly appreciate your advice.  

Thank you again for the useful package, sdmTMB.
Saang-Yoon

**Comment by ericward-noaa** [comment]
Yes, good question. The short answer is that we have not set up a smart way to detect folds as you suggest, so specifying them manually is your best option

**Reply by shyunuw** [reply]
Hi. 
Thanks for your prompt response. Cheers,
Saang-Yoon

**Comment by seananderson** [comment]
And by manually we mean use the `fold_ids` argument in `sdmTMB_cv`.

https://sdmtmb.github.io/sdmTMB/articles/cross-validation.html

**Reply by shyunuw** [reply]
Sean.  Many thanks for your attention and useful/helpful information. 
Saang-Yoon
